source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js

```diff
@@ -124,6 +124,40 @@ angular.module('locApp.modules.profile.controllers')
             },
             distance: '10'
         };
-    });
 
+        /**
+         * @ngdoc function
+         * @name checkID
+         * @description
+         * Check if the template id is unique
+         */
+        $scope.checkID = function() {
+            $scope.resourceForm.resourceId.$invalid = false;
+            var templateRefs = localStorageService.get('templateRefs');
+            templateRefs.find(function (t) {
+                if (t == $scope.resourceTemplate.id) {
+                    $scope.resourceForm.resourceId.$invalid = true;
+                };
+            });
+        };
 
+        /**
+         * @ngdoc function
+         * @name checkURI
+         * @description
+         * Check if URI resolves
+         */
+        $scope.checkURI = function() {
+            $scope.resourceForm.resourceURI.$warn = false;
+            var url = 'server/whichrt?uri=' + $scope.resourceTemplate.resourceURI;
+            $http({
+                method: 'HEAD',
+                url: url
+            })
+            .then(function (response) {
```

Comment:
This causes an exception when there is no response - "cannot read property match of undefined"